Universal Health script works, fully customizable, works on anything …

WarlockGames12 opened the pull request:
values works

Guidoge3 committed:
Universal Health script works, fully customizable, works on anything with a hp value needed

DaanKikkert commented on PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs:
```diff
@@ -0,0 +1,132 @@
+        Destroy(gameObject);
+    }
+
+    private void Update()
```
We willen geen update functie in het health script, aangezien dit op elke enemy runt is dit vreselijk voor performance. Ook is deze functionality niet iets wat we willen in het health script zelf maar op outside sources van regen.

DaanKikkert commented on PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs:
```diff
@@ -0,0 +1,132 @@
+    }
+    
+
+    private void OnCollisionEnter(Collision collision)
```
We doen collision checks niet in het health script zelf

DaanKikkert commented on PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs:
```diff
@@ -0,0 +1,132 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public enum ObjectType
```
Deze enum zou overbodig moeten zijn, we willen dat het health script universeel werkt zonder dat je vooraf moet aangeven op wat voor gameobject het zit

DaanKikkert commented on PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs:
```diff
@@ -0,0 +1,132 @@
+            _currentHealth = _maxHealth;
+    }
+
+    public void PassiveRegenHealth(int amount)
```
Hoewel ik passive health regen een goed idee vindt zou ik het niet in het health script zelf doen, aangezien we dit waarschijnlijk liever alleen op de speler hebben.

DaanKikkert commented on PVB Linx Groep 6/Assets/Code/Scripts/UniversalHealth.cs:
```diff
@@ -0,0 +1,132 @@
+        }
+    }
+
+    private IEnumerator DealingDamageDelay(int delayAmount)
```
Ik snap niet waar deze Enumerator voor is.

DaanKikkert committed:
Merge branch 'develop' into feature/universal-health

DaanKikkert reviewed (changes requested):
Check comments. Dit is een veel te groot script met veel redundant code, ik zal donderdag even direct meekijken en uitleg geven.

Guidoge3 committed:
Updated Universal Health script to be only for health from now on

regeneration script is split away from original commit and removed Enums for regen checks

DaanKikkert committed:
Merge branch 'develop' into feature/universal-health
